webapp/src/app/todo/todo.service.ts

```diff
@@ -0,0 +1,118 @@
+import { Injectable, OnDestroy } from '@angular/core';
+import { FireLoopRef, Todo } from '../shared/sdk/models';
+import { RealTime } from '../shared/sdk/services/core/real.time';
+import { Subscription } from 'rxjs/Subscription';
+import { UIService } from '../ui/ui.service';
+import { NgbModal, NgbModalRef } from '@ng-bootstrap/ng-bootstrap';
+
+@Injectable()
+export class TodoService implements OnDestroy {
+  // todos
+  private todo: Todo = new Todo();
+  private todos: Todo[] = new Array<Todo>();
+  private todoRef: FireLoopRef<Todo>;
+  // modal
+  private modalRef: NgbModalRef;
+  private modalTitle: string = '';
+  // subscriptions
+  private subscriptions: Subscription[] = new Array<Subscription>();
+
+  constructor(
+    private rt: RealTime,
+    private uiService: UIService,
+    private modal: NgbModal
+  ) {
+    this.subscriptions.push(
+      this.rt.onReady().subscribe(() => {
+        this.todoRef = this.rt.FireLoop.ref<Todo>(Todo);
+        this.subscriptions.push(this.todoRef.on('change').subscribe(
+          (todos: Todo[]) => {
+            this.todos = todos;
+          }));
+        this.subscriptions.push(this.todoRef.stats().subscribe(
+          (stats: any) => {
+            let data = new Array();
+            stats.forEach((stat: any) => {
+              data.push(stat.count);
+            });
+          }));
+      }));
+  }
+
+  ngOnDestroy() {
+    this.todoRef.dispose();
+    this.subscriptions.forEach(
+      (subscription: Subscription) => {
+        subscription.unsubscribe()
+      });
+  }
+
+  handleAction(event) {
+    switch (event.type) {
+      case 'create':
+        this.create(event.payload);
```

Comment:
When you `return this.create()` you don't need the break. It also make sure the switch statement does not 'fall through' to the statement below.